src/overseer/executor.clj

```diff
@@ -73,7 +73,7 @@
             (swap! ready-jobs disj job)
 
             (timbre/info (format "Reserving job %s (%s)" job-id (:job/type job)))
-            (if-let [reserved-job (core/reserve-job store job)]
+            (if-let [reserved-job (core/reserve-job store job-id)]
```

Comment:
Data structure name. I believe executors and some of the top-level stuff is missing tests. I'll try and add some, but by their nature they're kind of tricky side-effecting integration style tests